string/test/intersperse_test.cpp

```diff
@@ -87,10 +87,11 @@ void intersperse_fill_test()
 
 void intersperse_generate_test()
 {
-    string test = "test";
+    const string input = "test";
     CharGenerator g1, g2;
-    BOOST_CHECK(intersperse_generate_copy(test, g1) == "t1e2s3t");
+    BOOST_CHECK(intersperse_generate_copy(input, g1) == "t1e2s3t");
 
+    string test = "test";
```

Comment:
I like this - maybe just a naming nit but you could do something like `const string immutable_input = ...` and `string mutable_input = ...`